packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -163,6 +189,28 @@ export default createRule({
             },
           ],
         });
+      } else if (requireDefaultForNonUnion) {
+        const hasDefault = node.cases.some(
+          switchCase => switchCase.test == null,
+        );
+
+        if (!hasDefault) {
+          context.report({
+            node: node.discriminant,
+            messageId: 'switchIsNotExhaustive',
+            data: {
+              missingBranches: 'default',
+            },
+            suggest: [
+              {
+                messageId: 'addMissingCases',
+                fix(fixer): TSESLint.RuleFix | null {
```

Comment:
[Cleanup] There's no `| null` returned by this function or the others, so none of the three `| null`s need to be here.

```suggestion
                fix(fixer): TSESLint.RuleFix {
```